### PR TITLE
Add cop that checks for ApplicationMailer use

### PR DESCRIPTION
**Pull request by kphelps**
Based on `RailsApplicationModel` and `RailsApplicationSerializer`, this adds a cop to check that all mailers subclass `ApplicationMailer`

prime @tjwp

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # salsify_rubocop
 
+## v0.49.0
```
Comment (tjwp) [inline CHANGELOG.md]: Let's just bump this to `v0.48.1` -- it's unconventional, but what we've tried to do with this gem is keep the minor version in sync with rubocop.

File: config/default.yml [hunk 2]
```diff
@@ -2,6 +2,10 @@ Salsify/RailsApplicationSerializer:
   Description: 'Serializers must subclass ApplicationSerializer.'
   Enabled: false
 
+Salsify/RailsApplicationMailer:
+  Description: 'Models must subclass ApplicationMailer.'
```
Comment (tjwp) [inline config/default.yml]: "Mailers must ..."

File: lib/rubocop/cop/salsify/rails_application_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Salsify
+      # Check that mailers subclass ApplicationMailer with Rails 5.0
```
Comment (tjwp) [inline lib/rubocop/cop/salsify/rails_application_mailer.rb:8]: Is this a Rails 5 thing or was it introduced earlier?

File: spec/rubocop/cop/salsify/rails_application_mailer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  it "corrects nested mailers defined using Class.new" do
+    source = 'Nested::MyMailer = Class.new(ActionMailer::Base)'
+    inspect_source(cop, source)
+    expect(cop.messages).to eq(msgs)
+    expect(cop.highlights).to eq(highlights)
+    expect(autocorrect_source(cop, source)).to eq('Nested::MyMailer = Class.new(ApplicationMailer)')
+  end
+
+  it "correct anonymous mailers" do
```
Comment (tjwp) [inline spec/rubocop/cop/salsify/rails_application_mailer_spec.rb]: "corrects" (I know typo exists in the original too)

File: lib/salsify_rubocop/version.rb [hunk 5]
```diff
@@ -1,3 +1,3 @@
 module SalsifyRubocop
-  VERSION = '0.48.0'.freeze
+  VERSION = '0.49.0'.freeze
```
Comment (tjwp) [inline lib/salsify_rubocop/version.rb]: 0.48.1 here too

**Review by tjwp — APPROVED** [review]
In the process of submitting the ApplicationRecord cop to rubocop, was going to factor out an `EnforceSuperclass` mixin. But we can easily refactor what's here later if that lands.

File: CHANGELOG.md (hunk 1 above)
Comment (kphelps) [reply CHANGELOG.md]: makes sense

File: lib/rubocop/cop/salsify/rails_application_mailer.rb (hunk 3 above)
Comment (kphelps) [reply lib/rubocop/cop/salsify/rails_application_mailer.rb:8]: I think the pattern existed, but it only started being auto-generated in rails 5.

**Comment by tjwp** [comment]
@kphelps You may be interested in this: https://github.com/bbatsov/rubocop/pull/4279

**Pull request merged**